test/route-acl.test.js

```diff
@@ -0,0 +1,202 @@
+var routeAcl = require('../server/middleware/acl');
+var loopback = require('../index');
+
+var options = {
+  scopes: {
+    catalog: [{
+      'methods': 'GET',
+      'path': '/api/catalog'
+    }],
+    order: [{
+      methods: 'ALL',
+      path: '/api/orders'
+    }],
+    cancelOrder: [{
+      methods: 'delete',
+      'path': '/api/orders/:id'
+    }],
+    deleteEntities: [
+      {
+        methods: ['delete'],
+        path: '/api/:model'
+      }
+    ]
+  },
+  acls: [
+    {
+      role: '$everyone',
+      scopes: ['catalog'],
+      permission: 'ALLOW'
+    },
+    {
+      role: '$unauthenticated',
+      scopes: ['order'],
+      permission: 'DENY'
+    },
+    {
+      role: '$everyone',
+      scopes: ['deleteEntities'],
+      permission: 'DENY'
+    },
+    {
+      principalType: 'ROLE',
+      principalId: 'admin',
+      scopes: ['deleteEntities'],
+      permission: 'ALLOW'
+    },
+    {
+      principalType: 'ROLE',
+      principalId: 'cs',
+      scopes: ['cancelOrder'],
+      permission: 'ALLOW'
+    }
+  ],
+  roles: {
+    admin: [
+      {principalType: 'USER', principalId: 'mary'}
+    ],
+    demo: [
+      {principalType: 'USER', principalId: 'john'},
+      {principalType: 'APP', principalId: 'demo-app'}]
+  }
+};
+
+describe('route based ACLs', function() {
+  var handler;
+
+  before(function(done) {
+    handler = routeAcl(options);
+    var ds = loopback.createDataSource({
+      connector: 'memory',
+      name: 'db',
+      defaultForType: 'db'
+    });
+    loopback.Application.attachTo(ds);
```

Comment:
We shouldn't use this soon to be deprecated version of core models... Use the new `registry` approach.